src/common/util/jsonPatcher.js

```diff
@@ -273,7 +328,7 @@ define([
 
     function _getChangedNodesRec(patch, res, hash, gmePath) {
         var nodePatches = patch[hash] && patch[hash].patch, // Changes regarding node with hash
-            i,
+            i, j,
```

Comment:
Maybe we could generate all the changed node data at patch instead?
A `//TODO:...` would do for now.